### PR TITLE
added new (faster) duckdb (1.14s) and native python version

### PR DESCRIPTION
**Pull request by azrethos**
```
➜  python -mtimeit -n1 -r1 -t -s "from duck import main" "main()"

1 loop, best of 1: 1.14 sec per loop
```

### CODE REVIEW

**Comment by jinyus** [comment]
Hey, thanks for the PR. I'm willing to merge the duckdb version but it only outputs the id of the related posts which breaks compatibility with the other versions. 

The python version is slower than the current one `3.7s vs 1.8s` so I will keep the current impl.

File: python/related_duckdb.py [hunk 1]
```diff
@@ -19,18 +19,20 @@
...
+                    SELECT
+                        JSON_OBJECT('_id', rp._id, 'title', rp.title, 'tags', rp.tags)
+                    FROM related_counts rc
+                    JOIN posts rp ON rp._id = rc.related_post
+                    WHERE rc.source_post = p._id
+                    ORDER BY rc.relation_count DESC
+                    LIMIT 5
+                ) AS related
+            FROM posts p
+        ) TO '../related_posts_duckdb.jsonl';""",
```
Comment (azrethos) [inline python/related_duckdb.py:35]: technically this produces JSONL https://jsonlines.org/

Comment (azrethos) [reply python/related_duckdb.py:35]: `python -mtimeit -n1 -r1 -t -s "from duck import main" "main()"`


1 loop, best of 1: 1.98 sec per loop

**Comment by azrethos** [comment]
I went ahead and updated it.

Note, it produces jsonl which appears to be the superior format for this type of data 

https://jsonlines.org/

**Comment by jinyus** [comment]
Looks good

**Pull request merged**